cmip6_downscaling/disagg/terraclimate.py

```diff
@@ -314,27 +293,32 @@ def hydromod(
         aet = pet
         deficit = 0
 
-    excess = max(0, soil + delta_soil - awc)
-    excess_rain_only = max(0, soil + excess_after_liquid - awc)
-    if (demand < supply) and ((soil + delta_soil) > awc):
-        # if supply exceeds demand and the updated soil water is greater than
-        # the available capacity
-        runoff = excess
-        runoff_snow += excess - excess_rain_only
-        soil = awc
-    elif (demand < supply) and ((soil + delta_soil) <= awc):
-        # if supply exceeds demand and updated soil water
-        # is less than available water capacity you update
-        # the soil by the change in soil
-        soil += delta_soil
-        runoff = 0
+        if (soil + delta_soil) > awc:
+            # if supply exceeds demand and the updated soil water is greater than
+            # the available capacity
+            runoff = max(0, soil + delta_soil - awc)
+            runoff_snow = min(extra_runoff, melt)
+
+            rain_input = rain - extra_runoff - runoff_snow
+            excess_after_liquid = max(0, rain_input - pet)
+            excess_rain_only = max(0, soil + excess_after_liquid - awc)
+            runoff_snow += runoff - excess_rain_only
+            soil = awc
+        else:
+            # if supply exceeds demand and updated soil water
+            # is less than available water capacity you update
+            # the soil by the change in soil
+            soil += delta_soil
+            runoff = 0
+
+    # add the extra runoff component
     runoff += extra_runoff
 
     return {
         'aet': aet,
         'def': deficit,
         'q': runoff,
-        'snow_storage': snow_storage,
+        'swe': snow_storage,
```

Comment:
my only thought here (if we're being nitpick with the snow variable naming) is to just change `snow_storage` to `swe` throughout now. could be useful for readability so that our future selves aren't like "wait, `snow_storage`? is that different from `swe`?"